### PR TITLE
Always show debug bar in dev and add edit-entry link

### PR DESCRIPTION
**Pull request by davist11**

### CODE REVIEW

**Review by gregkohn — APPROVED** [review]
👏

File: modules/sample/src/Module.php [hunk 1]
```diff
@@ -25,13 +28,56 @@ public function init()
...
+                if (
+                    Craft::$app->config->general->devMode ||
+                    ($currentUser && $currentUser->can('accessCp'))
+                ) {
+                    $this->view->registerAssetBundle(Bundle::class);
+
+                    echo '<a
+                            href="' . $element->cpEditUrl . '"
+                            class="edit-entry"
+                            target="_blank"
```
Comment (olets) [inline modules/sample/src/Module.php:74]: is this negotiable? guess I can always tweak it ;P

File: modules/sample/src/Module.php [hunk 2]
```diff
@@ -25,13 +28,56 @@ public function init()
...
+    private function _bindEvents()
+    {
+        // Add edit entry link to front-end templates
+        Event::on(
+            View::class,
+            View::EVENT_END_BODY,
+            function(Event $e) {
+                $element = Craft::$app->getUrlManager()->getMatchedElement();
+                
+                if (!$element) return;
```
Comment (olets) [inline modules/sample/src/Module.php:61]: interesting what does this protect against? / what's the in-template equivalent of this check?

**Review by olets — APPROVED** [review]
Nice 🎉

File: modules/sample/src/Module.php (hunk 1 above)
Comment (davist11) [reply modules/sample/src/Module.php:74]: Opening in a new window seems like a good default to me. Certainly can modify it to be whatever you want once you've got it setup in your project.

File: modules/sample/src/Module.php (hunk 2 above)
Comment (davist11) [reply modules/sample/src/Module.php:61]: This makes sure that you are on a URL that references a Craft element.

Comment (olets) [reply modules/sample/src/Module.php:61]: Oh like for a route 👍

File: modules/sample/src/Module.php (hunk 1 above)
Comment (olets) [reply modules/sample/src/Module.php:74]: What about adding an "external link" arrow to the link for accessibility?

Comment (davist11) [reply modules/sample/src/Module.php:74]: Feels a bit like overkill IMO

**Pull request closed without merging**